winpr/libwinpr/input/scancode.c

```diff
@@ -114,81 +558,86 @@ DWORD GetVirtualKeyCodeFromVirtualScanCode(DWORD scancode, DWORD dwKeyboardType)
 	if (codeIndex > 127)
 		return VK_NONE;
 
-	if ((dwKeyboardType != WINPR_KBD_TYPE_IBM_ENHANCED) &&
-	    (dwKeyboardType != WINPR_KBD_TYPE_JAPANESE))
-		dwKeyboardType = WINPR_KBD_TYPE_IBM_ENHANCED;
+	switch (dwKeyboardType)
+	{
+			/* IBM and compatible */
+		case WINPR_KBD_TYPE_IBM_PC_XT:
+		case WINPR_KBD_TYPE_OLIVETTI_ICO:
+		case WINPR_KBD_TYPE_IBM_PC_AT:
+		case WINPR_KBD_TYPE_IBM_ENHANCED:
+			if (scancode & KBDMULTIVK)
+				return KBD4X_1[codeIndex];
+			if (scancode & KBDMULTIVK)
+				return KBD4X[codeIndex];
```

Comment:
```
			if (scancode & KBDMULTIVK)
				return KBD4X_1[codeIndex];
			if (scancode & KBDMULTIVK)
				return KBD4X[codeIndex];
```
Ah, the flag check does not look correct to me. Probably, that's why some keys don't work in my testing here.

Reply:
Upon checking the other conversion direction, I guess the latter check should use the `KBDEXT` flag.

Reply:
indeeed. fixed.